### DISCUSSION TITLE
getting event using aptos `getModuleEventsByEventType`

### DISCUSSION THREAD

**Discussion by Pelumi527** [aptos-labs/aptos-developer-discussions#277] [Questions] [answered]
### Discord user ID

pels.

### Describe your question in detail.

I trying to get the event using aptos `getModuleEventsByEventType` but I noticed that the events are arranged according to how it happens, I would like the get the latest event to use it on my app.

### What have you tried or looked at?  Or how can we reproduce the error?

To reproduce you can check the event of this module 
`0x939aa7c3d2f33607047b65b9ab27567e647fae07037f80f99bc6f2e598a58b97 ::nft_swap:: ListingCreatedEvent`

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xmaayan** [comment, marked as answer]
You should be able to do something like

```
const events = await aptos.getModuleEventsByEventType({
      eventType: "0x939aa7c3d2f33607047b65b9ab27567e647fae07037f80f99bc6f2e598a58b97 ::nft_swap:: ListingCreatedEvent",
      options: {
        orderBy: [{ creation_number: "desc" }],
      },
    });
```

**Reply by Pelumi527** [reply]
oh alright, thanks `creation_number` didn't sort according, but I was able to use the `transaction_version`
